scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/policy/license/LicenseSchedulingPolicy.java

```diff
@@ -186,17 +187,30 @@ private boolean isTaskExecutableUsingLicense(EligibleTaskDescriptor task) {
 
             // Do not execute if one of the required license can not be obtained
             final String[] requiredLicensesArray = requiredLicenses.split(",");
-            for (int i = 0; i < requiredLicensesArray.length; i++) {
+            final int nbRequiredLicenses = requiredLicensesArray.length;
+            boolean[] jobAlreadyUseLicense = new boolean[nbRequiredLicenses];
+
+            for (int i = 0; i < nbRequiredLicenses; i++) {
 
                 final String currentRequiredLicense = requiredLicensesArray[i];
+
+                // If the license is already obtained at the job level, use it without getting a new one
+                LinkedBlockingQueue<JobDescriptorImpl> eligibleJobsDescriptorsLicense = eligibleJobsDescriptorsLicenses.get(currentRequiredLicense);
+                if (containsJob(eligibleJobsDescriptorsLicense,
+                                ((EligibleTaskDescriptorImpl) task).getInternal().getJobId())) {
+                    jobAlreadyUseLicense[i] = true;
+                    continue;
+                }
```

Comment:
I think this block of code could be moved to second iteration. It would avoid to execute that block when returning false. You could merge the two conditions to return false only when it is required.

Reply:
We cannot shortcut the loop (return false if the task cannot take the current license) before testing before if  the job already got the current license. To improve the code readability, i removed the continue in the "if" and added a "else if" just after. But i decided to keep the result of the "if" (job already got the current license or not) for the next iteration, for not performing the test again before registering the task.

Reply:
Finally i can simplify it following your suggestions. Thx!